libs/datamodel/core/src/configuration/source/loader.rs

```diff
@@ -96,41 +105,69 @@ impl SourceLoader {
             ));
         }
 
-        for decl in &self.source_declarations {
-            // The provider given in the config block identifies the source type.
-            // TODO: The second condition is a fallback to mitigate the postgres -> postgresql rename. It should be
-            // renamed at some point.
-            if provider == decl.connector_type() || (decl.connector_type() == "postgresql" && provider == "postgres") {
-                let source = decl
-                    .create(
-                        source_name,
-                        StringFromEnvVar {
-                            from_env_var: env_var_for_url,
-                            value: url,
-                        },
-                        &ast_source.documentation.clone().map(|comment| comment.text),
-                    )
-                    .map_err(|err_msg| {
-                        DatamodelError::new_source_validation_error(&err_msg, source_name, url_args.span())
-                    });
-
-                return Ok(Some(source?));
-            }
+        let documentation = ast_source.documentation.clone().map(|comment| comment.text);
+        let url = StringFromEnvVar {
+            from_env_var: env_var_for_url,
+            value: url,
+        };
+
+        let all_datasource_providers: Vec<_> = providers
+            .iter()
+            .filter_map(|provider| self.get_datasource_provider(&provider))
+            .collect();
+
+        if all_datasource_providers.is_empty() {
+            return Err(DatamodelError::new_datasource_provider_not_known_error(
+                &providers.join(","),
+                provider_arg.span(),
+            ));
         }
 
-        Err(DatamodelError::new_source_not_known_error(
-            &provider,
-            provider_arg.span(),
-        ))
+        let validated_providers: Vec<_> = all_datasource_providers
+            .iter()
+            .map(|sd| {
```

Comment:
What does `sd` stand for here?

Reply:
That stood for `SourceDefinition` which was the old name for `DatasourceProvider`. Changing this to provider.